datasm/tools/tell_years_dsid.py

```diff
@@ -23,7 +23,8 @@ def assess_args():
 
     return args
 
-resource_path = '/p/user_pub/e3sm/staging/resource/'
+gp = os.environ['DSM_GETPATH']
```

Comment:
I suggest defining `DSM_GETPATH = os.environ["DSM_GETPATH"]` in a single file under `datasm/tools` and importing them in the Python files that need it. If you ever rename `DSM_GETPATH` (even if it is unlikely), you will need to find all instances of it across the Python tools and replace it.

`env.py`:
```python
import os

# DEFINE ENV VARS HERE
DSM_GETPATH = os.environ["DSM_GETPATH"]
```

`tell_years_dsid.py`:
```python
...
from datasm.tools.env import DSM_GETPATH
...
```

Reply:
I will study this.  The idea is to allow operators to redefine these "datasm system paths" without having to rebuild or re-install datasm code.  The "DSM_GETPATH" (for python modules) calls an external executable.  We could hide that in a single import module I guess.

So, instead of each module doing:

```
gp = os.environ['DSM_GETPATH']
wh_root = subprocess.run([gp, "STAGING_DATA"],stdout=subprocess.PIPE,text=True).stdout.strip()
pb_root = subprocess.run([gp, "PUBLICATION_DATA"],stdout=subprocess.PIPE,text=True).stdout.strip()
(etc etc etc as needed)
```

We obtain ALL "DSM_ROOT_PATHS" at once, and import the resulting array of paths(?)  e.g.:

from datasm.util import set_dsm_paths

```
dsm_paths = dsm_set_paths()
wh_root = dsm_paths['STAGING_DATA']
pb_root = dsm_paths['PUBLICATION_DATA']
(etc etc etc as needed)
```

Or is there an even better option?